flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/complex/ComplexIntegrationTest.java

```diff
@@ -0,0 +1,903 @@
+		expected.add("((534,628),97)");
+		expected.add("((535,631),98)");
+
+		StreamExecutionEnvironment env = new TestStreamEnvironment(1, 32);
+		env.setBufferTimeout(100);
+
+		env.addSource(new RectangleSource()).global()
+				.map(new RectangleMapFunction())
+				.window(Delta.of(0.0, new DeltaFunction<Tuple2<RectangleClass, Integer>>() {
+
+					@Override
+					public double getDelta(Tuple2<RectangleClass, Integer> oldDataPoint, Tuple2<RectangleClass,
+							Integer> newDataPoint) {
+						return (newDataPoint.f0.b - newDataPoint.f0.a) - (oldDataPoint.f0.b - oldDataPoint.f0.a);
+					}
+				}, new Tuple2<RectangleClass, Integer>(new RectangleClass(100, 100), 0)))
+				.mapWindow(new MyWindowMapFunction())
+				.flatten()
+				.addSink(new ToListSink<Tuple2<RectangleClass, Integer>>("test7"));
+
+		env.execute();
+
+		Collections.sort(expected);
+		Collections.sort(results.get("test7"));
+		assertEquals(expected, results.get("test7"));
+	}
+
+	@Test
+	public void complexIntegrationTest8() throws Exception {
+
+		ArrayList<String> expected = new ArrayList<String>();
+
+		StreamExecutionEnvironment env = new TestStreamEnvironment(1, 32);
+		env.setBufferTimeout(1);
+
+		SplitDataStream<Long> splittedStream1 = env.generateSequence(1, 10)
+				.map(new MapFunction<Long, Long>() {
+
+					@Override
+					public Long map(Long value) throws Exception {
+						return value;
+					}
+				})
+				.filter(new FilterFunction<Long>() {
+
+					@Override
+					public boolean filter(Long value) throws Exception {
+						return true;
+					}
+				})
+				.flatMap(new SquareFlatMapFunction())
+				.split(new OutputSelector<Long>() {
+
+					@Override
+					public Iterable<String> select(Long value) {
+						ArrayList<String> output = new ArrayList<String>();
+						if(value < 8) {
+							output.add("first");
+						} else {
+							output.add("second");
+						}
+						return output;
+					}
+				});
+
+		//splittedStream1.select("second").print();
+
+		SplitDataStream<Long> splittedStream2 = splittedStream1.select("first").split(new OutputSelector<Long>() {
+
+			@Override
+			public Iterable<String> select(Long value) {
+				ArrayList<String> output = new ArrayList<String>();
+				if(value < 5) {
+					output.add("third");
+				} else {
+					output.add("fourth");
+				}
+				return output;
+			}
+		});
+
+		DataStream<Long> dataStream3 = splittedStream1.select("second").map(new MapFunction<Long, Long>() {
+
+			@Override
+			public Long map(Long value) throws Exception {
+				return value;
+			}
+		});
+
+		dataStream3.print();
+		splittedStream2.select("third").print();
+		// BUG!!!
+		dataStream3.merge(splittedStream2.select("third")).print();
+
+		//splittedStream2.select("fourth").print();
+
+		//.addSink(new ToListSink<Long>("test8"));
+
+		env.execute();
+
+		//Collections.sort(expected);
+		//Collections.sort(results.get("test8"));
+		//assertEquals(expected, results.get("test8"));
+	}
+
+	public static class InnerPojo {
+		public Long f0;
```

Comment:
Flink should also support stuff like java collections, date-types etc.
We frequently see users on the ML using that stuff. If its easy to integrate into the test, I would suggest using such a type as well.